scripts/sqa_builds.py

```diff
@@ -0,0 +1,208 @@
+
+    for revision, build_details in state.items():
+        try:
+            build = sqa.get_build(build_details.get("uuid"))
+            if not build:
+                log.warning(f"Build with UUID {build_details.get("uuid")} not found.")
+                continue
+            results[revision] = {
+                "status": build.status,
+                "result": build.result,
+                "uuid": str(build.uuid),
+                "arch": build_details.get("arch"),
+                "base": build_details.get("base"),
+                "channel": build_details.get("channel")
+            }
+        except sqa.SQAFailure as e:
+            log.error(f"Failed to get build {build_details.get("uuid")}: {e}")
+
+    return results
+
+
+def create_one_build(
+    state: dict[str, dict], track: str, risk_level: str, arch: str, base: str, dry_run: bool
+):
+    """Process the given channel based on its current state."""
+    log.info(f"Current state: {state}")
+    channel = f"{track}/{risk_level}"
+    k8s_operator_bundle = charmhub.Bundle("k8s-operator")
+    for charm in ["k8s", "k8s-worker"]:
+        log.info(f"Getting revisions for {charm} charm on channel {channel}")
+        try:
+            revision_matrix = charmhub.get_revision_matrix(charm, channel)
+        except HTTPError:
+            log.exception(
+                f"failed to get revision matrix for charm {charm} channel {channel}"
+            )
+            return
+
+        if not revision_matrix:
+            log.exception(f"charm {charm} has no revisions on channel {channel}")
+            return
+
+        log.info(
+            f"Revision matrix for {charm} on channel {channel} \n: {revision_matrix}"
+        )
+        k8s_operator_bundle.set(charm, revision_matrix)
+
+    k8s_revision_matrix = k8s_operator_bundle.get("k8s")
+    testable_revisions = []
+    for matrix_base in k8s_revision_matrix.get_bases():
+        for matrix_arch in k8s_revision_matrix.get_archs():
+            if arch and arch != matrix_arch:
+                continue
+
+            if base and base != matrix_base:
+                continue
+
+            revision = k8s_revision_matrix.get(matrix_arch, matrix_base)
+            if revision and not state.get(str(revision)):
+                testable_revisions.append((matrix_base, matrix_arch))
+
+    if not testable_revisions:
+        log.info(
+            "The constraints resulted in no testable revisions or they are already tested. Skipping..."
+        )
+        return
+
+    log.info(
+        f"Found {len(testable_revisions)} testable revision(s) for channel {channel}: {testable_revisions}"
+    )
+    (base_in_test, arch_in_test) = random.choice(testable_revisions)        #nosec
+    log.info(f"Selected base {base_in_test} and arch {arch_in_test} for testing.")
+
+    revisions = k8s_operator_bundle.get_revisions(arch_in_test, base_in_test)
+    version = f"{revisions.get("k8s_revision")}-{arch_in_test}-{base_in_test}-{track}-{risk_level}"
```

Comment:
this is going to show up as the "addon" name?  Can we improve this?


```suggestion
    version = f"k8s-build-{revisions.get("k8s_revision")}-{arch_in_test}-{base_in_test}-{track}-{risk_level}"
```